### ISSUE TITLE
Upgrade Action clarification

### ISSUE THREAD

**Issue by a2life** [SharePoint/sp-dev-docs#4639]
When the spfx code is modified and uploaded to the appCatalog, the effect of this change is immediate in the site that the app is already deployed.  This "Upgrade Action" is only needed when it involves Assets modification. Although the notes implies this,  I am not sure if it is clearly stated.

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 2c30373b-3c44-c8bc-75d1-bb788efb479e
* Version Independent ID: 5ad284d8-3320-e7f3-0164-d29d3d82b74d
* Content: [Provision SharePoint assets from your SharePoint client-side web part](https://docs.microsoft.com/en-us/sharepoint/dev/spfx/web-parts/get-started/provision-sp-assets-from-package#feedback)
* Content Source: [docs/spfx/web-parts/get-started/provision-sp-assets-from-package.md](https://github.com/SharePoint/sp-dev-docs/blob/master/docs/spfx/web-parts/get-started/provision-sp-assets-from-package.md)
* Product: **sharepoint**
* Technology: **sharepoint-framework**
* GitHub Login: @spdevdocs
* Microsoft Alias: **spdevdocs**

**Comment by andrewconnell** [comment]
How would you change the page?

**Comment by a2life** [comment]
Sorry If this was not clear.  For brevity, lets say I start with 'HelloWorld" web part and already create a page that uses this web part.   Now I go back to the source and change html a bit.  like "Welcome to SharePoint!" to "SharePoint is Great!",  Change "version" in "package-solution.json" to "2.0.0.0",  package it up and  upload to AppCatalog and deploy it.   If I go back to the page and refresh,  it will show "SharePoint is Great!" So the code change is already in effect.  Installed version to this site is still Version "1.0.0.0" and version 2.0.0.0 is available (for upgrade) .  With this behavior and from the way it is written in this page, I concluded this Upgrade Action is mainly for Assets modification (like adding another list or modifying list structure) but code changes from TS source are immediate once you upload the solution and deploy / checked in in AppCatalog.

(come to think of it, there could be an issue if the modified source code depends on a list created from provisioning.  )
FYI, I am using Site Collection level App Catalog that we have created last year.   I am not sure if this makes any difference though.
